weave/tests/test_pantry_factory.py

```diff
@@ -0,0 +1,193 @@
+"""Pytest tests for the pantry factory."""
+import json
+import os
+import sys
+import tempfile
+
+import pytest
+
+import weave
+from weave import IndexSQLite
+from weave import IndexSQL
+from weave.pantry_factory import create_pantry
+from weave.index.index_pandas import IndexPandas
+from weave.tests.pytest_resources import PantryForTest, IndexForTest
+from weave.tests.pytest_resources import get_file_systems
+
+###############################################################################
+#                      Pytest Fixtures Documentation:                         #
+#            https://docs.pytest.org/en/7.3.x/how-to/fixtures.html            #
+#                                                                             #
+#                  https://docs.pytest.org/en/7.3.x/how-to/                   #
+#          fixtures.html#teardown-cleanup-aka-fixture-finalization            #
+#                                                                             #
+#  https://docs.pytest.org/en/7.3.x/how-to/fixtures.html#fixture-parametrize  #
+###############################################################################
+
+
+# Create fsspec objects to be tested, and add to file_systems list.
+file_systems, _ = get_file_systems()
+
+# Create Index CONSTRUCTORS of Indexes to be tested, and add to indexes list.
+indexes = [IndexPandas, IndexSQLite]
+indexes_ids = ["Pandas", "SQLite"]
+
+# Only add IndexSQL if the env variables are set and dependencies are present.
+if "WEAVE_SQL_HOST" in os.environ and "sqlalchemy" in sys.modules:
+    indexes.append(IndexSQL)
+    indexes_ids.append("SQL")
+
+# Create combinations of the above parameters to pass into the fixture.
+params = []
+params_ids = []
+for iter_file_system in file_systems:
+    for iter_index, iter_index_id in zip(indexes, indexes_ids):
+        params.append((iter_file_system, iter_index))
+        params_ids.append(f"{iter_file_system.__class__.__name__}-"
+                          f"-{iter_index_id}")
+
+@pytest.fixture(name="test_pantry", params=params, ids=params_ids)
+def fixture_test_pantry(request, tmpdir):
+    """Sets up test pantry for the tests."""
+    file_system = request.param[0]
+    pantry_path = (
+        "pytest-temp-pantry" f"{os.environ.get('WEAVE_PYTEST_SUFFIX', '')}"
+    )
+
+    test_pantry = PantryForTest(tmpdir, file_system, pantry_path=pantry_path)
+
+    index_constructor = request.param[1]
+    test_index = IndexForTest(
+        index_constructor=index_constructor,
+        file_system=file_system,
+        pantry_path=pantry_path,
+    )
+    index_name = test_index.index.__class__.__name__
+
+    yield test_pantry, index_name, index_constructor
+    test_pantry.cleanup_pantry()
+    test_index.cleanup_index()
+
+
+def test_pantry_factory_default_args(test_pantry):
```

Comment:
would it be redundant to make tests with bad variables that are not in the config file?